src/settings.rs

```diff
@@ -23,8 +23,7 @@ pub const NIX_TARBALL: &[u8] = include_bytes!(env!("NIX_INSTALLER_TARBALL_PATH")
 
 #[cfg(all(
     feature = "determinate-nix",
-    target_os = "linux",
-    not(target_arch = "x86")
+    not(all(target_os = "linux", target_arch = "x86"))
```

Comment:
Can we get a comment about this cfg? "If determinate-nix is enabled, and the target is not i686-linux, then embed the binary"

(It took me like 5 tries to understand what was going on here)